config/with.go

```diff
@@ -125,3 +125,25 @@ func WithSession(s auth.Session) HCPConfigOption {
 		return nil
 	}
 }
+
+// WithProfile is an option that can be used to provide a custom Profile struct.
+//
+// This should only be necessary for development purposes.
+func WithProfile(organizationID, projectID, hcpEnvironment string) HCPConfigOption {
```

Comment:
This seems like a good opportunity for using the Profile interface as a single parameter, so we don't have to keep updating this function signature every time we change what's in the profile.